api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -583,6 +583,10 @@ message HttpConnectionManager {
   // *not* the deprecated but similarly named :ref:`stream_error_on_invalid_http_messaging
   // <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.stream_error_on_invalid_http_messaging>`
   google.protobuf.BoolValue stream_error_on_invalid_http_message = 40;
+
+  // Optional maximum requests for a downstream connection.
+  // If not specified, there is no limit.
+  google.protobuf.UInt32Value max_requests_connection = 43;
```

Comment:
@ggreenway @mk46 I'd vote to move this to `HttpProtocolOptions` and deprecate it in `Cluster` (with appropriate redirect in comments), it logically belongs there.

Reply:
+1. cc @alyssawilk who has been working with this config a lot lately.